### PR TITLE
8335743: jhsdb jstack cannot print some information on the waiting thread

### PR DESCRIPTION
**Pull request by tkiriyama**
This bug was introduced by JDK-8284161. "Object.wait (long timeoutMillis)" was changed to call "Object.wait0 (long timeoutMillis)" in JDK-8284161.

When "jhdsb jstack" is executed, the stack and lock information are printed in "sun.jvm.hotspot.runtime.JavaVFrame.printLockInfo(PrintStream tty, int frameCount)". This method checks whether the method is java.lang.Object.wait (...) and then reports the waitstate only if the check passes.
https://github.com/openjdk/jdk/blob/7bc8f9c150cbf457edf6144adba734ecd5ca5a0f/src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/JavaVFrame.java#L103

```
      if (getMethod().getName().asString().equals("wait") &&
          getMethod().getMethodHolder().getName().asString().equals("java/lang/Object")) {
```

However, after JDK-8284161, the waiting thread waits on "java.lang.Object.wait0 (long timeoutMillis)", so it no longer reports the waitstate.

I changed "printLockInfo(PrintStream tty, int frameCount)" to check for "java.lang.Object.wait0 (...)".
I confirmed that the lock information is correctly printed with this fix.
I tested hotspot/jtreg/serviceability/sa/ and jdk/sun/tools/jhsdb/heapconfig/, and there were no regressions by this fix.

Would anyone review this change, please?

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found leading lowercase letter in issue title for `8335743: jhsdb jstack cannot print some information on the waiting thread`

### Issue
 * [JDK-8335743](https://bugs.openjdk.org/browse/JDK-8335743): jhsdb jstack cannot print some information on the waiting thread (**Bug** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20049/head:pull/20049` \
`$ git checkout pull/20049`

Update a local copy of the PR: \
`$ git checkout pull/20049` \
`$ git pull https://git.openjdk.org/jdk.git pull/20049/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20049`

View PR using the GUI difftool: \
`$ git pr show -t 20049`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20049.diff">https://git.openjdk.org/jdk/pull/20049.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20049#issuecomment-2210531207)

### CODE REVIEW

File: test/hotspot/jtreg/serviceability/sa/LingeredAppWithLock.java [hunk 1]
```diff
@@ -36,24 +36,37 @@ public static void lockMethod(Object lock) {
...
+            }
+        }
+    }
+
     public static void main(String args[]) {
         Thread classLock1 = new Thread(() -> lockMethod(LingeredAppWithLock.class));
         Thread classLock2 = new Thread(() -> lockMethod(LingeredAppWithLock.class));
         Thread objectLock = new Thread(() -> lockMethod(classLock1));
         Thread primitiveLock = new Thread(() -> lockMethod(int.class));
+        Thread objectWait = new Thread(() -> waitMethod(new Object()));
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/serviceability/sa/LingeredAppWithLock.java:54]: Use of `new Object()` could be problematic here as the JIT can recognise that this is a non-escaping object and elide the synchronization.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Good catch on the underlying issue. I think the test needs an adjustment though.

Thanks

**Review by plummercj — APPROVED** [review]
Looks good. Thank you for adding the test.

**Review by kevinjwalls — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]
Update looks good. Thanks

**Comment by tkiriyama** [comment]
Thank you all for the reviews. 

/integrate

**Pull request closed without merging**